app/models/container_template.rb

```diff
@@ -34,11 +34,23 @@ def instantiate(params, project = nil)
                                             :namespace => project
                                           },
                                           :objects    => objects,
-                                          :parameters => params)
+                                          :parameters => params_to_hashes(params))
     create_objects(processed_template['objects'], project)
     @created_objects.each { |obj| obj[:miq_class] = MIQ_ENTITY_MAPPING[obj[:kind]] }
   end
 
+  def params_to_hashes(params)
+    params.collect do |param|
+      {
+        :name     => param.name,
+        :value    => param.value,
+        :generate => param.generate,
+        :from     => param.from,
+        :required => param.required
+      }
+    end
+  end
+
```

Comment:
`hashes` (plural!) is widely used in refresh to mean array of hashes.  Though yes, in `ems_inv_to_hashes` it means hash of arrays of hashes :-|

How about `template_params_to_hashes` ?

Why `each_with_object([])`, won't 
```
params.collect do |param|
  .param.attributes.slice("name", "value", "generate", "from", "required")
end
```
work?

@zakiva note this slice returns string keys and you had symbols.